### ISSUE TITLE
Conditionnal Isdroppable between 3 columns

### ISSUE THREAD

**Issue by Eric-Juquel** [atlassian/react-beautiful-dnd#1913]
### Description

Hi , i'm new on react and beautiful dnd that is a great library and maybe i'm missing something in the doc about IsDroppable prop but in my case i got 3 columns : A:manual messages, B:results, C:Antenne .
 Items from A & B are coming from back-end and we should be able to drag&drop them in C to be published . the way is we should be able to drop them back to theyre departure column only , i mean items comming from A should only be droppable in C or A and items from B in C or B but an item coming originaly from A should never go to B and inverse(sry for my bloody english ...). so i give an attribute type to each items but i don't find the right way to use it . 
Could somebody help me to fix that please ? 
https://codesandbox.io/s/keen-colden-6itoo

**Comment by imjoshellis** [comment]
I'm new to this lib, so this might not be the most performant solution, but I set up a sandbox with this functionality here: https://codesandbox.io/s/clever-roentgen-40vlr?file=/src/App.js

It uses `onDragStart` to grab the `draggableId` and set the correct columns to disabled/enabled.

Let me know if you have any questions about the code. I can explain in more detail later.

**Comment by Eric-Juquel** [comment]
Great solution , works perfectly and easy to understand , Thanks @imjoshellis .
 i got a new issue with this lib cause i'm now using data from the API , so i call API onDragend to switch the message from source column to destination column but have a 0.2 sec "flickering" where the item appears quickly back in source before going in destination . 
Even sometimes i need to drag the item 2 times to be effective that is not expecting behavior ... i can't send it here cause it's a professional project but i will try to reproduce on sandbox .
 I'm using Context to set the global state and a reducer , maybe should i go to Redux to manage my API request ?

**Comment by imjoshellis** [comment]
@Eric-Juquel I'm using Redux and had the same kind of flickering issues. 

My solution was to move the task in the frontend state while waiting on the API call's Promise to resolve. When resolved, I verify the position of the task moved correctly in the DB (and if it didn't, I fix the frontend state). I don't think Redux is necessary.
